google/cloud/spanner/internal/connection_impl_test.cc

```diff
@@ -345,62 +350,37 @@ TEST(ConnectionImplTest, ReadTooManyTransientFailures) {
-                spanner_proto::BatchCreateSessionsRequest const& request) {
-            EXPECT_EQ(db.FullName(), request.database());
-            return MakeSessionsResponse({"test-session-name"});
-          });
-
+  EXPECT_CALL(*mock, BatchCreateSessions(_, HasDatabase(db)))
+      .WillOnce(Return(MakeSessionsResponse({"test-session-name"})));
   EXPECT_CALL(*mock, StreamingRead(_, _))
       .Times(AtLeast(2))
-      .WillRepeatedly(
-          [](grpc::ClientContext&, spanner_proto::ReadRequest const&) {
-            auto reader = absl::make_unique<MockGrpcReader>();
-            EXPECT_CALL(*reader, Read(_)).WillOnce(Return(false));
-            EXPECT_CALL(*reader, Finish())
-                .WillOnce(Return(
-                    grpc::Status(grpc::StatusCode::UNAVAILABLE, "try-again")));
-            return reader;
-          });
+      // This won't compile without `Unused` despite what the gMock docs say.
+      .WillRepeatedly([](Unused, Unused) {
+        return MakeFailingReader({grpc::StatusCode::UNAVAILABLE, "try-again"});
+      });
 
   auto rows =
       conn->Read({MakeSingleUseTransaction(Transaction::ReadOnlyOptions()),
                   "table",
                   KeySet::All(),
                   {"UserId", "UserName"}});
   for (auto& row : rows) {
-    EXPECT_EQ(StatusCode::kUnavailable, row.status().code());
-    EXPECT_THAT(row.status().message(), HasSubstr("try-again"));
+    EXPECT_THAT(row,
+                StatusIs(StatusCode::kUnavailable, HasSubstr("try-again")));
   }
 }
 
 /// @test Verify implicit "begin transaction" in Read() works.
 TEST(ConnectionImplTest, ReadImplicitBeginTransaction) {
   auto mock = std::make_shared<spanner_testing::MockSpannerStub>();
-
   auto db = Database("dummy_project", "dummy_instance", "dummy_database_id");
   auto conn = MakeConnection(
       db, {mock}, ConnectionOptions{grpc::InsecureChannelCredentials()});
-  EXPECT_CALL(*mock, BatchCreateSessions(_, _))
-      .WillOnce(
-          [&db](grpc::ClientContext&,
-                spanner_proto::BatchCreateSessionsRequest const& request) {
-            EXPECT_EQ(db.FullName(), request.database());
-            return MakeSessionsResponse({"test-session-name"});
-          });
-
-  auto grpc_reader = absl::make_unique<MockGrpcReader>();
+  EXPECT_CALL(*mock, BatchCreateSessions(_, HasDatabase(db)))
+      .WillOnce(Return(MakeSessionsResponse({"test-session-name"})));
   auto constexpr kText = R"pb(metadata: { transaction: { id: "ABCDEF00" } })pb";
-  spanner_proto::PartialResultSet response;
-  ASSERT_TRUE(TextFormat::ParseFromString(kText, &response));
-  EXPECT_CALL(*grpc_reader, Read(_))
-      .WillOnce(DoAll(SetArgPointee<0>(response), Return(true)))
-      .WillOnce(Return(false));
-  EXPECT_CALL(*grpc_reader, Finish()).WillOnce(Return(grpc::Status()));
   EXPECT_CALL(*mock, StreamingRead(_, _))
-      .WillOnce(Return(ByMove(std::move(grpc_reader))));
+      .WillOnce(Return(ByMove(MakeReader({kText}))));
```

Comment:
I am not sure if this is expected, but the for-loop below is empty, the code coverage things `EXPECT_STATUS_OK()` goes untouched.